doc/cookbook/source/examples/classifier/lmnn.rst

```diff
@@ -0,0 +1,47 @@
+=================================
+Largest Margin Nearest Neighbours
+=================================
+
+Largest Margin Nearest Neighbours is a metric learning algorithm.  It learns a metric that can be used with the K-Nearest neigbour algorithm.
+
+The Mahalanobis distance metric which is obtained is computed for  samples :math:`x_i` and :math:`x_j` as:
```

Comment:
I think LMNN always returns `CustomMahalanobisDistance`. http://www.shogun-toolbox.org/doc/en/latest/classshogun_1_1CLMNN.html#a9f96703addfc52317b1f2bb97e20a7ee

I have mentioned it as sgclass now.